### ISSUE TITLE
[ZS] logic_pickups, weaponrequiredforammo 1, isn't aware of deployables which use the ammo type

### ISSUE THREAD

**Issue by BoXiousNet** [JetBoom/noxiousnet-issues#137]
If a player has a deployable in their inventory which uses the ammo type, they'll find they are unable to pick the prop_ammo boxes.

Affected items:
- Gunturret (SMG ammo)
- Blast Turret (Buckshot ammo)
- Zapper (Pulse ammo)
- Drone (SMG ammo)
- Repair field emitter (Pulse ammo)

**Comment by Raox2** [comment]
Any map to easily replicate this on?

**Comment by BClark09** [comment]
Just create a prop_ammo yourself?

```lua
ent = ents.Create("prop_ammo")
if ent:IsValid() then
	ent:SetAmmoType("pulse")
	ent:SetAmmo(20)
	ent:SetPos(Entity(1):GetPos()) --you
	ent:SetAngles(Entity(1):SetAngles())
	ent:Spawn()
end
```

**Comment by Raox2** [comment]
Well I mean a map with weaponrequiredforammo but I'll guess I'll make that.

**Comment by BClark09** [comment]
Set `GAMEMODE.WeaponRequiredForAmmo`, there's no need to use logic_pickups.

**Comment by Raox2** [comment]
I had no idea this functionality existed in that form, actually. Reading over it, it makes more sense to classify this as an oversight/enhancement rather than a bug, assign it to use resupply ammo types, and make sure those weapons without one have one.

**Comment by BClark09** [comment]
I agree, but the reason why this happens is because deployables have their own ammotype, so you wouldn't be able to give them a new one. The "enhancement" for inventory should be simple:

- Give each deployable weapon an ent.TurretAmmoType (= "pulse" for example). (Some have this value already)
- When prop_ammo is checking for "hasweapon", check for 
```lua
ent.DeployAmmoType and string.lower(self:GetAmmoType())==string.lower(ent.TurretAmmoType)
```
However, what happens if someone has already deployed the entity? Shouldn't we allow them to pick up the ammo for it?

**Comment by Raox2** [comment]
I've got the first part done, but I need some way to determine what deployables a player owns in an efficient way.

**Comment by BoXiousNet** [comment]
zs_obj_dump_v12, zs_obj_countrytrain_v4, zs_obj_bluevelvet_v4, all recently updated maps make use of this.
